### ISSUE TITLE
utility functions and decorators

### ISSUE THREAD

**Issue by kdschlosser** [OpenZWave/python-openzwave#162]
@bibi21000 
I put together some utility functions for logging. and for deprecated.

the logging functions are to be used as a decorator. they log data being passed to and or returned from a function. they are still a rough draft and I am going to be adding where the call to the function was made from. This will be a handy thing when trying to debug a data path issue.

I made up a deprecated decorator/function as well. It looks as tho you started working on making one but it was not complete. The one that I wrote handles symbols methods functions classes and properties. To make the deprecated decorator work correctly with symbols required me to use property in order to produce the warning only when the symbol gets used. and making the deprecated decorator work correctly with properties was a challenge as well. I wrote the function so you can decorate the set and the get independently so if you only wanted to deprecate a portion of it that can be done as well.

I wanted you to have a look see at it. you need to run the demonstration.py file 

[utilities.zip](https://github.com/OpenZWave/python-openzwave/files/3015699/utilities.zip)

**Comment by bibi21000** [comment]
Hi,
Sorry for the late answer.
Nice idea for the deprecated decorator. Will be very usefull for the rewrite of the command classes. Maybe in 0.5 we should let the old command classes and mark them as deprecated. And add the new ones too. And in 0.6 reove the old ones.
And the log decorator should be very usefull too for testinf/developing the new command classes.
I'm ok for both
Thank for your work

**Comment by kdschlosser** [comment]
Hey no worries. I have greatly improved upon them already. I added the file and line number of where the call was made from as well as the file and line number of where the item is.

I also wrote a little GUI for parsing a log file. it uses wxPython and should be cross platform for mac linux and windows. it displays the code from the files for you and show the call arguments as well as the returned. makes it easy to read the log file information.
